GrocifyApp.BLL/Implementations/ProductMeasureService.cs

```diff
@@ -0,0 +1,27 @@
+﻿using GrocifyApp.BLL.Data.Consts.ENConsts;
+using GrocifyApp.BLL.Interfaces;
+using GrocifyApp.DAL.Exceptions;
+using GrocifyApp.DAL.Models;
+using GrocifyApp.DAL.Repositories.Interfaces;
+
+namespace GrocifyApp.BLL.Implementations
+{
+    public class ProductMeasureService : EntitiesService<ProductMeasure>, IProductMeasureService
+    {
+        public ProductMeasureService(IRepository<ProductMeasure> repository) : base(repository)
+        {
+        }
+
+        public override async Task Insert(ProductMeasure entity, CancellationTokenSource? token = null)
+        {
+            try
+            {
+                await base.Insert(entity, token);
+            }
+            catch (Microsoft.EntityFrameworkCore.DbUpdateException)
+            {
+                throw new CustomException(string.Format(GenericConsts.Exceptions.EntityExistsFormat, GenericConsts.Entities.ProductMeasure));
```

Comment:
Another suggestion: why don't we put this try catch exception in generic? and we can have a virtual string property, by default is "Entity", but it can be override and each service decides what to call it.
here we just need to override the property, not the entire method